terraform/routines.tf

```diff
@@ -173,3 +173,43 @@ resource "google_bigquery_routine" "user_agent_parser" {
   }
 EOF
 }
+
+resource "google_bigquery_routine" "parse_furnished_types" {
+  dataset_id   = local.routines_dataset
+  routine_id   = "parse_furnished_types${local.branch_suffix_underscore_edition}"
+  routine_type = "SCALAR_FUNCTION"
+  language     = "JAVASCRIPT"
```

Comment:
Why is this done with Javascript instead of SQL?

Reply:
I did not see a way of doing this kind of logic in SQL, care to provide a pseudo example?